kong/plugins/hmac-auth/access.lua

```diff
@@ -29,6 +31,63 @@ local SIGNATURE_NOT_SAME = "HMAC signature does not match"
 
 local _M = {}
 
+local hmac = {
+  ["hmac-sha1"] = function(secret, data)
+    return crypto.hmac.digest("sha1", data, secret, true)
+  end,
+  ["hmac-sha256"] = function(secret, data)
+    return crypto.hmac.digest("sha256", data, secret, true)
+  end,
+  ["hmac-sha384"] = function(secret, data)
+    return crypto.hmac.digest("sha384", data, secret, true)
+  end,
+  ["hmac-sha512"] = function(secret, data)
+    return crypto.hmac.digest("sha512", data, secret, true)
+  end
+}
+
+local function list_as_set(list)
+  local set = {}
+  for _, v in ipairs(list) do
+    set[v] = true;
+  end
+
+  return set
+end
+
+local function validate_params(params, conf)
+  -- check username and signature are present
+  if not (params.username and params.signature) then
+    return nil, "username or signature missing"
+  end
+
+  -- check enforced headers are present
+  local enfoprced_header_set = list_as_set(conf.enforce_headers)
+  for _, header in ipairs(params.hmac_headers) do
+    enfoprced_header_set[header] = nil
+  end
+
+  for _, header in ipairs(conf.enforce_headers) do
+    if enfoprced_header_set[header] then
+      return nil, "enforced header not used for signature creation"
+    end
+  end
+
+  -- check supported alorithm used
+  local matched = false
+  for _, algo in ipairs(conf.algorithms) do
+    if algo == params.algorithm then
+      matched = true
```

Comment:
lets just return true, no need to keep running through this loop once we have a match? also means we dont need `matched`:

```
for _, algo in ipairs(conf.algorithms) do
  if algo == params.algorithm then
    return true
  end
end

return nil, fmt("algorithm %s not supported", params.algorithm)
```